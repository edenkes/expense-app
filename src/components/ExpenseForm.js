import React from "react";
import moment from "moment";
import {SingleDatePicker} from "react-dates";
import 'react-dates/lib/css/_datepicker.css'

export default class ExpenseForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount / 100).toString() : '',
            createdAt: props.expense ? props.expense.createdAt : moment(),
            buttonText: props.buttonText,
            error: '',
            calendarFocused: false
        }
    }

    onDescriptionChange = (e) => {
        const description = e.target.value
        this.setState({description})
    }
    onNoteChange = (e) => {
        const note = e.target.value
        this.setState({note})
    }
    onAmountChange = (e) => {
        const amount = e.target.value
        if (amount.match(/^\d{0,10}(\d\.\d{0,2})?$/)) {
            this.setState({amount})
        }
    }
    onDateChange = (createdAt) => {
        if (createdAt){
            this.setState({createdAt})

        }
    }
    onFocusChange = ({focused}) => {
        this.setState({calendarFocused: focused})
    }
    onSubmit = (e) => {
        e.preventDefault()

        if (!this.state.description || !this.state.amount){
            this.setState({error: "must give description and amount!"})
        }else {
            this.setState({error: ""})
        }

        this.props.onSubmit({
            description: this.state.description,
            note: this.state.note,
            amount: parseFloat(this.state.amount) * 100,
            createdAt: this.state.createdAt,
        })
    }
    render() {
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    <label >description</label>
                    <input type="text"
                           placeholder="description"
                           id="description"
                           value={this.state.description}
                           onChange={this.onDescriptionChange}
                    /><br/>

                    <label >note</label>
                    <textarea placeholder="note" id="note"
                              value={this.state.note}
                              onChange={this.onNoteChange}/><br/>

                    <label >amount</label>
                    <input type="text" placeholder="amount" id="amount"
                           value={this.state.amount}
                           onChange={this.onAmountChange}

                    /><br/>


                    <SingleDatePicker
                    date={this.state.createdAt}
                    onDateChange={this.onDateChange}
                    focused={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={()=>false}
                    /><br/>
                    <button type="submit" value="Submit">{this.state.buttonText}</button>
                </form>
                <div>
                    {this.state.error}
                </div>
            </div>
        )
    }
}