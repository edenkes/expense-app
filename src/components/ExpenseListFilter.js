import React from "react";
import {connect} from "react-redux";
import {setEndDate, setStartDate, setTextFilter, sortByAmount, sortByDate} from "../actions/filters";
import {DateRangePicker} from "react-dates";

class ExpenseListFilter extends React.Component {
    state = {
        calendarFocused: null
    }

    onDatesChange = ({startDate, endDate}) => {
        console.log(startDate)
        this.props.dispatch(setStartDate(startDate))
        this.props.dispatch(setEndDate(endDate))

    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({
            calendarFocused
        }));
    }

    render() {
        return (
            <div>
                <input type="text"
                       placeholder="Filter text"
                       value={this.props.filters.text}
                       onChange={(e) => {
                           this.props.dispatch(setTextFilter(e.target.value))
                       }}/>

                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
                <select id="sortBy"
                        value={this.props.filters.sortBy}
                        onChange={(e) => {
                            if (e.target.value === "amount") {
                                this.props.dispatch(sortByAmount())
                            } else {
                                this.props.dispatch(sortByDate())
                            }
                        }}>
                    <option value="amount">Amount</option>
                    <option value="date">Date</option>
                </select>
            </div>

        )
    }
}

const mapStateToProp = (state) => ({
    filters: state.filters
})

export default connect(mapStateToProp)(ExpenseListFilter);