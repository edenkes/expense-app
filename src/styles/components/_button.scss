.button {
  background: $light-blue;
  //background: dodgerblue;
  border: none;
  color: aliceblue;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}