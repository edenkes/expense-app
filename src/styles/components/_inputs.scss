.test-input {
  border: 1px solid #cacccd;
  height: 50px;
  font-size: $font-size-large;
  font-weight: 300;
  padding: $s-size;
}


.select {
  @extend .test-input;
}

.textarea {
  @extend .test-input;
  height: 10rem;
}