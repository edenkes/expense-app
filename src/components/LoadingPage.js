import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {startAddExpense} from '../actions/expenses';

const LoadingPage = () => (
    <div className={"loader"}>
        <img className={"loader__image"} src={"/images/loader.gif"}/>
    </div>
)
export default LoadingPage;