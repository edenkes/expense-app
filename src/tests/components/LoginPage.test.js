import React from 'react';
import {shallow} from 'enzyme';
import {LoginPage} from "../../components/LoginPage";
import {Header} from "../../components/Header";

test('should correctly render Login Page', () => {
    const wrapper = shallow(<LoginPage startLogin={()=> {}}/>);
    expect(wrapper).toMatchSnapshot();
})


test('should call startLogin on button click ', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<Header startLogin={startLogin()} />)
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled()
})

