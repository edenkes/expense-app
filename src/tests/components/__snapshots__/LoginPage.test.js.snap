// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Login Page 1`] = `
<div>
  <h1>
    Login Page
  </h1>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
